blocks/simple-cards/simple-cards.js

```diff
@@ -0,0 +1,57 @@
+import {
+  div, domEl, a, span,
+} from '../../scripts/dom-helpers.js';
+
+export default function decorate(block) {
+  const dom = domEl('ul');
+
+  [...block.children].forEach((row) => {
+    const cardPicture = row.children[0].querySelector('picture');
+
+    if (row.children[1].querySelectorAll('p').length === 1) { /* simple cards overlay with no icons */
```

Comment:
Rather than checking for content in the rows, you can add variants to a block by putting qualifiers in `()` after the block name

In this case, we should use `(icon)` as a qualifier to signal that the block has icon content.

This would change your CSS dramatically - but significantly reduce your JS logic. Don't break apart any content from this cell, just use it as-is and style it with a parent qualifier of `.block.simple-cards.icon`